arches/app/media/js/viewmodels/map-editor.js

```diff
@@ -5,19 +5,22 @@ define([
     'knockout',
     'knockout-mapping',
     'uuid',
-    'mapbox-gl-draw',
     'geojson-extent',
     'geojsonhint',
     'togeojson',
     'proj4',
     'views/components/map',
     'views/components/cards/select-feature-layers',
     'text!templates/views/components/cards/map-popup.htm'
-], function(arches, $, _, ko, koMapping, uuid, MapboxDraw, geojsonExtent, geojsonhint, toGeoJSON, proj4, MapComponentViewModel, selectFeatureLayersFactory, popupTemplate) {
+], function(arches, $, _, ko, koMapping, uuid, geojsonExtent, geojsonhint, toGeoJSON, proj4, MapComponentViewModel, selectFeatureLayersFactory, popupTemplate) {
     var viewModel = function(params) {
         var self = this;
         var padding = 40;
         var drawFeatures;
+        self.MapboxDraw = undefined;
```

Comment:
No; no need for self - I'll fix that.  In re: to your latter question - unless I'm misunderstanding you, I can't change the interior of the viewmodel constructor function to the mapbox-gl-require block because the require is asynchronous - so when the template is renderered, mapbox-gl-draw is not ready yet, so none of the needed observables are setup on the viewmodel and the template loading process errors out.  Happy to discuss this further and come up with an alternative loading method.